src/solidlsp/companion_ls.py

```
@@ -0,0 +1,389 @@
+        failed_cleanups: list[tuple[str, str]] = []
+        for uri in set(self._indexed_file_uris):
+            for companion in self._companions.values():
+                try:
+                    if uri in companion.open_file_buffers:
+                        file_buffer = companion.open_file_buffers[uri]
+                        file_buffer.ref_count -= 1
+
+                        if file_buffer.ref_count == 0:
+                            companion.server.notify.did_close_text_document({"textDocument": {"uri": uri}})
+                            del companion.open_file_buffers[uri]
+                except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)